src/csharp/Microsoft.Spark.E2ETest/UdfTests/UdfSimpleTypesTests.cs

```diff
@@ -166,5 +167,51 @@ public void TestUdfWithReturnAsTimestampType()
                 }
             }
         }
+
+        /// <summary>
+        /// UDFs defined in separate threads to validate ThreadStatic bug fix.
+        /// </summary>
+        [Fact]
+        public void TestUdfWithMultipleThreads()
+        {
+            try
+            {
+                static void Method1()
+                {
+                    try
+                    {
+                        Func<Column, Column> udf1 = Udf<string, string>(
+                        str => $"{str} - test udf1");
+                    }
+                    catch (Exception e)
+                    {
+                        throw new Exception(e.ToString());
+                    }
+                }
+
+                static void Method2()
+                {
+                    try
+                    {
+                        Func<Column, Column> udf2 = Udf<string, string>(
+                        str => $"{str} - test udf2");
+                    }
+                    catch (Exception e)
+                    {
+                        throw new Exception(e.ToString());
+                    }
+                }
+
+                Thread t1 = new Thread(Method1);
+                Thread t2 = new Thread(Method2);
+                t1.Start();
+                t1.Join();
+                t2.Start();
+            }
+            catch (Exception e)
+            {
+                Assert.NotNull(e);
```

Comment:
Can we make sure this is the exception we threw in our logic (add a simple exception string).

Reply:
Sure, however this is the case for verifying the test fails with ThreadStatic. For the bug fix and actual test to check-in, I will have to assert the exception is null.

Reply:
Yes.